apps/web/src/services/workspace-home/components/BookmarkHeader.vue

```diff
@@ -23,13 +29,52 @@ const bookmarkState = bookmarkStore.state;
 
 const { width } = useWindowSize();
 
+const componentRef = ref<HTMLElement|null>(null);
+const toolboxRef = ref<HTMLElement|null>(null);
+const folderItemsRef = ref<HTMLElement[]|null>(null);
+
+const moreButtonRef = ref<HTMLElement | null>(null);
+const moreContextMenuRef = ref<any|null>(null);
+
+const { width: containerWidth } = useElementSize(componentRef);
+const { width: toolboxWidth } = useElementSize(toolboxRef);
+const { top: moreButtonTop, height: moreButtonHeight } = useElementBounding(moreButtonRef);
+
 const storeState = reactive({
-    filterByFolder: computed<string|undefined>(() => bookmarkState.filterByFolder),
+    filterByFolder: computed<string|undefined|TranslateResult>(() => bookmarkState.filterByFolder),
     isFullMode: computed<boolean>(() => bookmarkState.isFullMode),
     isFileFullMode: computed<boolean>(() => bookmarkState.isFileFullMode),
 });
 const state = reactive({
     isMobileSize: computed<boolean>(() => width.value < screens.mobile.max),
+    folderListMaxWidth: computed<number>(() => containerWidth.value - 150 - toolboxWidth.value),
```

Comment:
nitpick: What is the value of 150 here? It would be nice if you could express it as a constant!